src/test/key_tests.cpp

```diff
@@ -188,4 +188,14 @@ BOOST_AUTO_TEST_CASE(key_signature_tests)
     BOOST_CHECK(found_small);
 }
 
+BOOST_AUTO_TEST_CASE(key_key_negation)
+{
+    CKey key = DecodeSecret(strSecret1C);
```

Comment:
The test here could be a bit more extensive (for example, signing/verifying a fixed message with the corresponding private key, and checking that the private key roundtrips after 2 negations).

Reply:
Good point. Extended the test to cover the double negation signature comparison.